test/runtests.jl

```diff
@@ -141,6 +141,11 @@ end
     @test mysum2(x, y) == x + y
     @check_allocs (x::Bar)(y::Bar) = x.val + y.val
     @test Bar(x)(Bar(y)) == x + y
+
+    # Callsite forms
+    @test 1 + x == @check_allocs 1 + x
+    @test x^2 == @check_allocs (a -> a^2)(x)
+    @test_throws AllocCheck.AllocCheckFailure @check_allocs same_ccall()
```

Comment:
In the spirit of your recent issues, I double-checked and I don't think this handles kwargs correctly right:
```julia
julia> foo(a,b;c=1) = a + b + c
foo (generic function with 1 method)
julia> @check_allocs foo(1,2;c=50)
4
julia> foo(1,2;c=50)
53
```

Reply:
Thanks for doing this check! I'm currently trying to work out why exactly this is happening. With `@macroexpand` and a `@show` invocation, this is what I see with your example:

```julia
@show local variables
ex = :(foo(1, 2; c = 30))
mod = Main
source = :(#= REPL[16]:1 =#)
args = Any[:($(Expr(:parameters, :($(Expr(:kw, :c, 30)))))), 1, 2]
args_template = Any[:($(Expr(:parameters, :c))), Symbol("##arg#255"), Symbol("##arg#256")]
passthrough_defun = :(function (var"##arg#255", var"##arg#256"; c)
      foo(var"##arg#255", var"##arg#256"; c)
  end)
original_fn = :(function (var"##arg#255", var"##arg#256", c;)
      foo(var"##arg#255", var"##arg#256"; )
  end)
f_sym = Symbol("##fn_alias#257")
wrapper_fn = :(function ($(Expr(:escape, :(var"##arg#255"::Any))), $(Expr(:escape, :(var"##arg#256"::Any))); $(Expr(:escape, :c)))
      #= REPL[16]:1 =#
      callable_tt = Tuple{map(Core.Typeof, ($(Expr(:escape, Symbol("##arg#255"))), $(Expr(:escape, Symbol("##arg#256"))), $(Expr(:escape, :c))))...}
      #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:157 =#
      callable = (AllocCheck.compile_callable)(var"##fn_alias#257", callable_tt; ignore_throw = true)
      #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:158 =#
      if length(callable.analysis) > 0
          #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:159 =#
          throw(AllocCheckFailure(callable.analysis))
      end
      #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:161 =#
      callable($(Expr(:escape, Symbol("##arg#255"))), $(Expr(:escape, Symbol("##arg#256"))), $(Expr(:escape, :c)))
  end)

@macroexpand
quote
    #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:188 =#
    let var"#98###fn_alias#257" = function (var"##arg#255", var"##arg#256", c;)
                foo(var"##arg#255", var"##arg#256"; )
            end
        #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:189 =#
        var"#99###alloccheck_fn#260" = function (var"##arg#255"::Any, var"##arg#256"::Any; c)
                #= REPL[16]:1 =#
                var"#102#callable_tt" = AllocCheck.Tuple{AllocCheck.map((AllocCheck.Core).Typeof, (var"##arg#255", var"##arg#256", c))...}
                #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:157 =#
                var"#103#callable" = (AllocCheck.compile_callable)(var"#98###fn_alias#257", var"#102#callable_tt"; ignore_throw = true)
                #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:158 =#
                if AllocCheck.length((var"#103#callable").analysis) > 0
                    #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:159 =#
                    AllocCheck.throw(AllocCheck.AllocCheckFailure((var"#103#callable").analysis))
                end
                #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:161 =#
                var"#103#callable"(var"##arg#255", var"##arg#256", c)
            end
        #= /home/tec/.julia/dev/AllocCheck/src/macro.jl:190 =#
        var"#99###alloccheck_fn#260"(1, 2; c = 30)
    end
end
```

Reply:
To be clear, I'm a bit stuck on this, and help from someone else would be good to move this forward.